### PR TITLE
Turtles!

### PR DESCRIPTION
**Pull request by sadenar**
#### Summary
Content "Adds Turtles, Snapping Turtles and Tortoises to the game, as well as mutation lines for them"

#### Purpose of change
I was thinking about what I wanted to add next, and I realized we did not have any turtles in game, despite them being an actually pretty abundant animal in New England. As turtles are extremely cool critters, this could not stand and I thus decided to add snapping turtles, aquatic turtles and tortoises to the game along with mutation lines for them.

#### Describe the solution
I quickly created a small explanation of the 12 monsters I'm planning to add:
Tortles

This should explain roughly what I'm going for, turtles would be going for a "weird mutated shell that hurts to look at and can cut you badly" vibe and have pretty high armor stats, snapping turtles would be very scary to fight in melee, but lackluster in defense compared to the other two, and tortoises would become the tankiest of tanks, but lack in offensive capabilities.
All 3 are turtles, and as such will have horrendous speed stats but pretty high armor and HP, allowing them to tank damage but be easily evaded, so don't expect to be outchased by a turtle unless you're a Rabbit mutant. :D

Both turtle lines have pretty meh vision, as quick searches online have shown me that them being aquatic means their vision out of it is pretty bad. Tortoises however will be given pretty high vision, they're no eagles, but they're supposed to be among the best seeing reptiles.

Aggressiveness wise, I would be more going with snapping turtles being on sight murder machines while the other two types will be pretty passive, only actively hunting you as megas.

I may need some help or feedback on the aquatic turtle, as I'm not exactly sure yet if a debuff attack players/npcs the closer and better able to discern its shell's details you are would be balanced or even feasible currently, if not I could change its theme to "hedgehog turtle".

So far I'm done with the following things I've identified as needing doing:
- [ ] Add a turtle body type
- [x] Add aquatic turtles and their mutant evolutions
- [ ] Add the special attacks of the aquatic turtles
- [x] Add tortoises and their mutant evolutions
- [x] Add snapping turtles and their mutant evolutions
- [ ] Add cool shells to butcher them for, and since this requires new harvest categories, add mutant anatomy snippets for them
- [ ] Maybe add cool stuff to use the shells for?

#### Describe alternatives you've considered
- Going the Karol fish route and adding every species of turtle found in most of NE, decided to settle with two types for terrestrial vs aquatic turtles, as well as snapping turtles because I couldn't just ignore them.
- Only adding one line of turtles in this PR, decided making a more massive one where all turtlely stuff was nicely packed together was a better idea.

#### Testing
None yet, I'm just publishing this along with my single commit so far to be able to go to sleep.

#### Additional context
Turtles are pretty awesome.

### CODE REVIEW

File: data/json/monsters/reptile_amphibian.json [hunk 1]
```diff
@@ -416,6 +416,160 @@
...
+    "vision_night": 7,
+    "harvest": "mammal_tiny",
+    "dissect": "dissect_lizard_sample_single",
+    "anger_triggers": [ "HURT" ],
+    "fear_triggers": [ "PLAYER_CLOSE" ],
+    "placate_triggers": [ "PLAYER_WEAK" ],
+    "reproduction": { "baby_monster": "mon_turtle_s", "baby_count": 15, "baby_timer": 420 },
+    "baby_flags": [ "SUMMER" ],
+    "//babies": "hard to determine an egg number for now with a generic freshwater turtle, as some species will lay hundreds and others only a couple.",
+    "upgrades": { "half_life": 34, "into": "mon_turtle_big" },
```
Comment (Venera3) [inline data/json/monsters/reptile_amphibian.json:454]: Please don't use half-life for mutant animals. It's one of the things setting them apart from zombies.

File: data/json/monsters/reptile_amphibian.json [hunk 2]
```diff
@@ -416,6 +416,160 @@
...
+    "flags": [ "SEES", "HEARS", "SMELLS", "KEENNOSE", "CLIMBS", "PATH_AVOID_DANGER_1", "ANIMAL", "SWIMS" ],
+    "armor": { "bash": 5, "cut": 5, "stab": 5, "heat": 2, "bullet": 4, "electric": 1, "acid": 1 }
+  },
+  {
+    "id": "mon_turtle_big",
+    "type": "MONSTER",
+    "name": { "str": "luminary's vision" },
```
Comment (Venera3) [inline data/json/monsters/reptile_amphibian.json:461]: This is way too purple - you need to ask yourself if it's clear for the player who just got hit with the safe mode popup what you're talking about: is this a ghost? a mirror? a gracken memory? portal storm nasty?

File: data/json/monsters/reptile_amphibian.json [hunk 3]
```diff
@@ -416,6 +416,160 @@
...
+    "material": [ "flesh" ],
+    "symbol": "T",
+    "color": "green",
+    "aggression": 40,
+    "morale": 80,
+    "aggro_character": false,
+    "melee_skill": 7,
+    "melee_dice": 3,
+    "melee_dice_sides": 6,
+    "melee_damage": [ { "damage_type": "cut", "amount": 8 } ],
```
Comment (Venera3) [inline data/json/monsters/reptile_amphibian.json:558]: If you want them to bite, add a bite attack - I'd suggest a grabbing bite + drag.

**Review by Venera3 — COMMENTED** [review]
I'm not very hot on the turtles becoming always angry murdermons if you want to do more predatory species anyway - a huge herbivore that weirds you out if you look at it too much is already interesting.

The names are also a readability problem, especially if you want to do three different variants - is a vision bigger than a mirror or a maze? There is also the option of having these be one-two nether mutants instead of reproducing normal ones that are inexplicably are magic.

**Comment by Karol1223** [comment]
> weird mutated shell that hurts to look at

This really doesn't feel mutant but nether. Mutants don't really go the way of breaking the laws of physics or random psychic powers, I specifically distinguished our dogs/cats from this by separating them into a different category of nether-touched animals.

**Comment by fairyarmadillo** [comment]
The psychic turtle line might be a better fit for Xedra Evolved or MoM.

File: data/json/monsters/reptile_amphibian.json (hunk 1 above)
Comment (sadenar) [reply data/json/monsters/reptile_amphibian.json:454]: Most of what I used was copied over from the rattlesnake line, what would be the ideal thing to use instead?

File: data/json/monsters/reptile_amphibian.json (hunk 2 above)
Comment (sadenar) [reply data/json/monsters/reptile_amphibian.json:461]: Forgot to add turtle at the end of that string, but from the feedback so far I'm considering just scrapping the weird mind effects of the aquatic turtle and turning them into spikies.

File: data/json/monsters/reptile_amphibian.json (hunk 3 above)
Comment (sadenar) [reply data/json/monsters/reptile_amphibian.json:558]: Will do, those turtles I don't want focused on their bites, but they would have one, how would I go about them being able to slam you with a spiked shell?

**Comment by sadenar** [comment]
> > weird mutated shell that hurts to look at
> 
> This really doesn't feel mutant but nether. Mutants don't really go the way of breaking the laws of physics or random psychic powers, I specifically distinguished our dogs/cats from this by separating them into a different category of nether-touched animals.
> 
> Also, I'm with Venera, these names are horrendous for anyone meeting them for the first time ingame. Don't go with poetic names, name them what people in the Cataclysm would name them so that the person playing actually knows what they're working with. This naming scheme is a disaster.

I would agree on both naming and theming for the first line I've added so far, which I'm considering just scrapping altogether from this feedback, but I would disagree that the other two lines do not show a pretty non-poetic but clear rising intensity line on the "more tanky/bitey" characteristics they would have. Not every single mutant creature needs to be called big or giant X.

> The psychic turtle line might be a better fit for Xedra Evolved or MoM. Lots of people play with those and more content would certainly be welcome.

Point taken, I might add this to a follow up PR or add it to this PR in time, MoM sounds like the better fit if Standing is fine with it.

**Comment by Standing-Storm** [comment]
> Point taken, I might add this to a follow up PR or add it to this PR in time, MoM sounds like the better fit if Standing is fine with it.

I would love having psionic animals added in.  It's something that should exist but which I haven't had any time (or inspiration) to add.

**Comment by Termineitor244** [comment]
I'm happy to see turtles coming to the Cataclysm!! That said, I too think the picnic turtle should be in the mentioned mods rather than mainline if it is to be added as a "natural" evolution rather than as a nether monster or a weird variant grown in a lab...

The other 2 mutation lines are pretty standard for me, though I think it would be nice to have more possible mutation paths for these monsters, but that can always be done later.

Speaking of psionic animals, it would be nice to have psionic spiders too! A new faction of psionic spiders would be pretty cool... But well, maybe in another time.

**Comment by sadenar** [comment]
Added the snapping turtles, the unmutated version is mostly where I want it, wanted to add them before going to sleep, I'll fix the parameters of the mutant snappers sometime this week, then focus on giving them proper attacks as well as weakpoints, finishing up all of it with 3 egg items, harvest groups, items and snippets.

**Comment by sadenar** [comment]
Added Harvest Groups. Harvest snippets, shell and skull items as well as special attacks remaining to do, as well as turtle weakpoints.

**Pull request closed without merging**